api/src/search/backend/load_opportunities_to_index.py

```diff
@@ -346,18 +346,29 @@ def get_attachment_json_for_opportunity(
 
         attachments = []
         for att in opp_attachments:
-            if self.filter_attachment(att):
-                with file_util.open_stream(
+            if not self.filter_attachment(att):
+                self.increment(self.Metrics.ATTACHMENTS_UNPROCESSED)
+                continue
+            try:
+                file_text = extract_text_from_file(
                     att.file_location,
-                    "rb",
-                ) as file:
-                    file_content = file.read()
-                    attachments.append(
-                        {
-                            "filename": att.file_name,
-                            "data": base64.b64encode(file_content).decode("utf-8"),
-                        }
-                    )
+                    char_limit=TEXT_EXTRACTOR_CHAR_LIMIT,
+                    raise_on_error=True,
+                )
+                attachments.append(
+                    {
+                        "filename": att.file_name,
+                        "attachment": {
+                            "content": file_text,
+                        },
+                    }
+                )
+                self.increment(self.Metrics.ATTACHMENTS_PROCESSED)
+            except Exception as e:
+                self.increment(self.Metrics.ATTACHMENTS_UNPROCESSED)
+                logger.warning(
+                    "text-extractor: error extracting text", extra={"err": e, "attachment": att}
```

Comment:
Could we put the attachment ID instead (+ add opportunity ID while we're at it)? I think what this would do at the moment is look like (with some random mock data locally): `<OpportunityAttachment(attachment_id=7,opportunity_id=113,file_location='s3://local-mock-public-bucket/opportunities/113/attachments/83321342/whole.txt',mime_type='model/x3d+vrml',file_name='whole.txt' ...)>`

This wouldn't fit too well into the JSON object/be searchable.